### PR TITLE
fix: Single State for Button Disable and Loading

### LINKED ISSUE #711: [🕹️] Loading on Login/Signup: Disable Buttons
### What side quest or challenge are you solving?

On the Login/Signup page, when any authentication button is clicked, all other buttons remain enabled, but the entire form should be disabled.

Current Behaviour:

https://github.com/user-attachments/assets/0764253c-f526-440d-9ba0-300b478c3c7d

How it should be:

https://github.com/user-attachments/assets/25cd52a9-d359-40a5-86b8-121a2dbcc976

### Points

?

### Description

I will disable all the buttons when any one of them is clicked.

### Provide proof that you've completed the task

I will assign myself and start working.

### PR DESCRIPTION
**Pull request by mohdjami**
fixes #711  
### Description
Created a single state for both loading and disabling buttons.
This code not only adds a single state to the Login page but also reduces the use of separate loaders without affecting any error handling in the form. No need to maintain separate states for each button.

### Used states before changes
```typescript

  const [isLoginWithEmail, setIsLoginWithEmail] = useState<boolean>(false);
  const [isLoginWithGoogle, setIsLoginWithGoogle] = useState<boolean>(false);
  const [isLoginWithLinkedIn, setIsLoginWithLinkedIn] =
    useState<boolean>(false);
``` 

### State after changes
```typescript
 const [clickedMethod, setClickedMethod] = useState<AuthMethod | undefined>(
    undefined,
  );
``` 

### Example button before changes
```typescript
          <Button
              onClick={() => {
                setIsLoginWithLinkedIn(true);
                signIn("linkedin", {
                  ...(next && next.length > 0 ? { callbackUrl: next } : {}),
                }).then((res) => {
                  if (res?.status) {
                    setIsLoginWithLinkedIn(false);
                  }
                });
              }}
              disabled={isLoginWithLinkedIn}
              className="flex items-center justify-center space-x-2 border border-gray-200 bg-gray-100 font-normal text-gray-900 hover:bg-gray-200"
            >
              {isLoginWithLinkedIn ? (
                <Loader className="mr-2 h-5 w-5 animate-spin " />
              ) : (
                <LinkedIn />
              )}
              <span>Continue with LinkedIn</span>
            </Button>
``` 
### Example button after changes
```typescript
          <Button
              onClick={() => {
                setClickedMethod("google");
                signIn("google", {
                  ...(next && next.length > 0 ? { callbackUrl: next } : {}),
                }).then((res) => {
                  if (res?.status) {
                    setClickedMethod(undefined);
                  }
                });
              }}
              loading={clickedMethod === "google"}
              disabled={clickedMethod && clickedMethod !== "google"}
              className="flex items-center justify-center space-x-2 border border-gray-200 bg-gray-100 font-normal text-gray-900 hover:bg-gray-200"
            >
              <Google className="h-5 w-5" />
              <span>Continue with Google</span>
            </Button>
``` 

### Preview
https://github.com/user-attachments/assets/64af5fac-8389-4dc0-b71a-1c94ed9271ab

This improves user experience by preventing multiple actions and maintaining a clear visual indication of the active button.

Final:

https://github.com/user-attachments/assets/e0d20dba-fae8-4f07-a36e-28fa0dd31f9e

### CODE REVIEW

**Comment by mohdjami** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by HarshTiwari-err** [comment]
@mohdjami 
Please test one scenario: 
Enter the email and click the submit button.
Check if your email is stuck on loading if there is any sort of server error.

**Comment by mohdjami** [comment]
> @mohdjami Please test one scenario: Enter the email and click the submit button. Check if your email is stuck on loading if there is any sort of server error.

Sure will do that, and try.

**Comment by mohdjami** [comment]
https://github.com/user-attachments/assets/de1cf45a-3567-4c12-a917-592f46ad0881

I tested with email and google login. It works fine, Let me know if you find anything wrong. My goal in this PR was not to change the existing error handling but just to fix the loading and disabling.

**Review by mfts — APPROVED** [review]
Great job!

File: app/(auth)/login/page.tsx [hunk 1]
```diff
@@ -17,20 +16,21 @@ import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
 import { Label } from "@/components/ui/label";
 
+export const authMethods = ["google", "email", "linkedin", "passkey"] as const;
+
+export type AuthMethod = (typeof authMethods)[number];
+
```
Comment (mfts) [inline app/(auth)/login/page.tsx]: @mohdjami you cannot export this from a page.tsx in app router. 

instead just add them as normal `const` inthe page function 

here's the build error: 
```
Type error: Page "app/(auth)/login/page.tsx" does not match the required types of a Next.js Page.
  "authMethods" is not a valid Page export field.
```

**Review by mfts — CHANGES_REQUESTED** [review]

File: app/(auth)/login/page.tsx (hunk 1 above)
Comment (mohdjami) [reply app/(auth)/login/page.tsx]: @mfts Fixed it.

File: app/(auth)/login/page.tsx [hunk 2]
```diff
@@ -17,20 +16,21 @@ import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
 import { Label } from "@/components/ui/label";
 
+const authMethods = ["google", "email", "linkedin", "passkey"] as const;
+
+type AuthMethod = (typeof authMethods)[number];
+
```
Comment (mfts) [inline app/(auth)/login/page.tsx]: @mohdjami can you put these inside the `export default function Login()` function

**Review by mfts — CHANGES_REQUESTED** [review]

File: app/(auth)/login/page.tsx (hunk 2 above)
Comment (mohdjami) [reply app/(auth)/login/page.tsx]: Done.

**Comment by mohdjami** [comment]
@mfts Thanks for addressing all the issues. Let me know if I missed anything.

**Pull request merged**